### PR TITLE
sharding-jdbc and sharding-proxy use same metaData refresh strategy.

### PR DESCRIPTION
**Pull request by yu199195**
i find sharding-jdbc and sharding-proxy have same code for metaDataRefresh after execute ddl.

so i remove MetaDataRefreshStrategy  from Sharding-jdbc-core to ShardingSphere-common.

then sharding-jdbc ,sharding-proxy can use same code.

### CODE REVIEW

File: shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/metadata/refresh/MetaDataRefreshStrategy.java [hunk 1]
```diff
@@ -15,26 +15,27 @@
...
+ *
  * @param <T> type of SQL statement context
  */
 public interface MetaDataRefreshStrategy<T extends SQLStatementContext> {
     
     /**
      * Refresh meta data.
      *
-     * @param shardingRuntimeContext sharding runtime context
+     * @param metaData sharding sphere meta data
```
Comment (kimmking) [inline shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/metadata/refresh/MetaDataRefreshStrategy.java:35]: sharding sphere only one word

File: shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/metadata/refresh/MetaDataRefreshStrategy.java [hunk 2]
```diff
@@ -15,26 +15,27 @@
...
 
-import org.apache.shardingsphere.shardingjdbc.jdbc.core.context.ShardingRuntimeContext;
 import org.apache.shardingsphere.sql.parser.binder.statement.SQLStatementContext;
+import org.apache.shardingsphere.underlying.common.metadata.ShardingSphereMetaData;
 
 import java.sql.SQLException;
 
 /**
  * Meta data refresh strategy.
- *
```
Comment (kimmking) [inline shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/metadata/refresh/MetaDataRefreshStrategy.java:27]: keep original

File: shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/AlterTableStatementMetaDataRefreshStrategyTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        AlterTableStatement alterTableStatement = new AlterTableStatement(new SimpleTableSegment(new TableNameSegment(1, 3, new IdentifierValue("t_order"))));
+        AlterTableStatementContext alterTableStatementContext = new AlterTableStatementContext(alterTableStatement);
+        metaDataRefreshStrategy.refreshMetaData(getMetaData(), alterTableStatementContext, tableName -> new TableMetaData(
+                Collections.singletonList(new ColumnMetaData("order_id", 1, "String", true, false, false)),
+                Collections.singletonList(new IndexMetaData("index_alter"))));
+        assertThat(getMetaData().getSchema().get("t_order").getIndexes().containsKey("index_alter"), is(true));
+    }
+}
+
+
```
Comment (kimmking) [inline shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/AlterTableStatementMetaDataRefreshStrategyTest.java:52]: one more line

File: shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/CreateTableStatementMetaDataRefreshStrategyTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        CreateTableStatement createTableStatement = new CreateTableStatement(new SimpleTableSegment(new TableNameSegment(1, 3, new IdentifierValue("t_order_0"))));
+        CreateTableStatementContext createTableStatementContext = new CreateTableStatementContext(createTableStatement);
+        metaDataRefreshStrategy.refreshMetaData(getMetaData(), createTableStatementContext, tableName -> new TableMetaData(
+                Collections.singletonList(new ColumnMetaData("order_id", 1, "String", true, false, false)),
+                Collections.singletonList(new IndexMetaData("index"))));
+        assertThat(getMetaData().getSchema().containsTable("t_order_0"), is(true));
+    }
+}
+
+
```
Comment (kimmking) [inline shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/CreateTableStatementMetaDataRefreshStrategyTest.java:52]: one more line

Comment (yu199195) [reply shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/CreateTableStatementMetaDataRefreshStrategyTest.java]: fix it

File: shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/metadata/refresh/MetaDataRefreshStrategy.java (hunk 1 above)
Comment (yu199195) [reply shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/metadata/refresh/MetaDataRefreshStrategy.java]: fix it

File: shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/metadata/refresh/MetaDataRefreshStrategy.java (hunk 2 above)
Comment (yu199195) [reply shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/metadata/refresh/MetaDataRefreshStrategy.java:27]: There's an extra space，now it not need

File: shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/AlterTableStatementMetaDataRefreshStrategyTest.java (hunk 3 above)
Comment (yu199195) [reply shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/AlterTableStatementMetaDataRefreshStrategyTest.java]: fit it

File: shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/CreateTableStatementMetaDataRefreshStrategyTest.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    public void refreshMetaData() {
+        MetaDataRefreshStrategy<CreateTableStatementContext> metaDataRefreshStrategy = new CreateTableStatementMetaDataRefreshStrategy();
+        CreateTableStatement createTableStatement = new CreateTableStatement(new SimpleTableSegment(new TableNameSegment(1, 3, new IdentifierValue("t_order_0"))));
+        CreateTableStatementContext createTableStatementContext = new CreateTableStatementContext(createTableStatement);
+        metaDataRefreshStrategy.refreshMetaData(getMetaData(), createTableStatementContext, tableName -> new TableMetaData(
+                Collections.singletonList(new ColumnMetaData("order_id", 1, "String", true, false, false)),
+                Collections.singletonList(new IndexMetaData("index"))));
+        assertThat(getMetaData().getSchema().containsTable("t_order_0"), is(true));
+    }
+}
```
Comment (kimmking) [inline shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/CreateTableStatementMetaDataRefreshStrategyTest.java:50]: let an empty line.

File: shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/AlterTableStatementMetaDataRefreshStrategyTest.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    public void refreshMetaData() {
+        MetaDataRefreshStrategy<AlterTableStatementContext> metaDataRefreshStrategy = new AlterTableStatementMetaDataRefreshStrategy();
+        AlterTableStatement alterTableStatement = new AlterTableStatement(new SimpleTableSegment(new TableNameSegment(1, 3, new IdentifierValue("t_order"))));
+        AlterTableStatementContext alterTableStatementContext = new AlterTableStatementContext(alterTableStatement);
+        metaDataRefreshStrategy.refreshMetaData(getMetaData(), alterTableStatementContext, tableName -> new TableMetaData(
+                Collections.singletonList(new ColumnMetaData("order_id", 1, "String", true, false, false)),
+                Collections.singletonList(new IndexMetaData("index_alter"))));
+        assertThat(getMetaData().getSchema().get("t_order").getIndexes().containsKey("index_alter"), is(true));
+    }
+}
```
Comment (kimmking) [inline shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/AlterTableStatementMetaDataRefreshStrategyTest.java:50]: let an empty line.

File: shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/metadata/refresh/MetaDataRefreshStrategy.java [hunk 7]
```diff
@@ -15,26 +15,27 @@
...
      *
-     * @param shardingRuntimeContext sharding runtime context
+     * @param metaData ShardingSphere meta data
      * @param sqlStatementContext SQL statement context
+     * @param callback callback
      * @throws SQLException SQL exception
      */
-    void refreshMetaData(ShardingRuntimeContext shardingRuntimeContext, T sqlStatementContext) throws SQLException;
+    void refreshMetaData(ShardingSphereMetaData metaData, T sqlStatementContext, TableMetaDataLoaderCallback callback) throws SQLException;
 }
```
Comment (kimmking) [inline shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/metadata/refresh/MetaDataRefreshStrategy.java:41]: let an empty line.

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/executor/AbstractStatementExecutor.java [hunk 8]
```diff
@@ -150,7 +155,19 @@ private void refreshMetaDataIfNeeded(final ShardingRuntimeContext runtimeContext
...
+        int maxConnectionsSizePerQuery = connection.getRuntimeContext().getProperties().<Integer>getValue(ConfigurationPropertyKey.MAX_CONNECTIONS_SIZE_PER_QUERY);
+        boolean isCheckingMetaData = connection.getRuntimeContext().getProperties().<Boolean>getValue(ConfigurationPropertyKey.CHECK_TABLE_METADATA_ENABLED);
+        TableMetaData result = new ShardingMetaDataLoader(connection.getDataSourceMap(), shardingRule, maxConnectionsSizePerQuery, isCheckingMetaData).load(tableName, databaseType);
+        result = new ShardingTableMetaDataDecorator().decorate(result, tableName, shardingRule);
+        if (!shardingRule.getEncryptRule().getEncryptTableNames().isEmpty()) {
+            result = new EncryptTableMetaDataDecorator().decorate(result, tableName, shardingRule.getEncryptRule());
+        }
+        return result;
+    }
 }
```
Comment (kimmking) [inline sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/executor/AbstractStatementExecutor.java:173]: let an empty line.

File: shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/AlterTableStatementMetaDataRefreshStrategyTest.java (hunk 6 above)
Comment (yu199195) [reply shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/AlterTableStatementMetaDataRefreshStrategyTest.java:50]: fix

File: shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/CreateTableStatementMetaDataRefreshStrategyTest.java (hunk 5 above)
Comment (yu199195) [reply shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/metadata/refresh/CreateTableStatementMetaDataRefreshStrategyTest.java:50]: this alreay empty line in the end file.

File: shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/metadata/refresh/MetaDataRefreshStrategy.java (hunk 7 above)
Comment (yu199195) [reply shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/metadata/refresh/MetaDataRefreshStrategy.java:41]: fix

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/executor/AbstractStatementExecutor.java (hunk 8 above)
Comment (yu199195) [reply sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/executor/AbstractStatementExecutor.java:173]: fix

**Pull request closed without merging**